identity-account/src/storage/memstore.rs

```diff
@@ -109,6 +113,32 @@ impl Storage for MemStore {
     }
   }
 
+  async fn key_insert(&self, id: IdentityId, location: &KeyLocation, secret_key: SecretKey) -> Result<PublicKey> {
+    let mut vaults: RwLockWriteGuard<'_, _> = self.vaults.write()?;
+    let vault: &mut MemVault = vaults.entry(id).or_default();
+
+    match location.method() {
+      MethodType::Ed25519VerificationKey2018 => {
+        let mut secret_key_bytes: [u8; 32] = secret_key.as_ref().try_into().expect("expected a slice of 32 bytes");
```

Comment:
Nit: return a proper `Error` instead of panicking.

E.g.
```
    let mut secret_key_bytes: [u8; 32] = <[u8; 32]>::try_from(secret_key.as_ref())
      .map_err(|err| Error::InvalidSecretKey(format!("expected a slice of 32 bytes - {}", err)))?;
```
This would require a new error type or one could wrap the `TryFromSliceError` directly. Replacing `try_into` with `try_from` is optional, I just prefer the latter to lookup which implementation it's using more easily.